src/core/Tools.cpp

```diff
@@ -240,4 +246,110 @@ void setupSearchPaths()
 #endif
 }
 
+//
+// Prevent memory dumps without admin privileges.
+// MiniDumpWriteDump function requires PROCESS_QUERY_INFORMATION and PROCESS_VM_READ
+// see: https://msdn.microsoft.com/en-us/library/windows/desktop/ms680360%28v=vs.85%29.aspx
+//
```

Comment:
Add the following to this comment:

>If you deny to open the process PROCESS_QUERY_INFORMATION and PROCESS_VM_READ access rights, you can't dump memory anymore. We do this using discretionary access control list (DACL) on Windows.
>
>This works as long as you do not have admin privileges, since then you are able to grant yourself the SE_TAKE_OWNERSHIP_NAME or SE_DEBUG_NAME privilege and circumvent the DACL.
>
>This function grants the process minimal access rights and denies everything else. Effectively this prevents memory dumps on windows and disallows other processes from accessing our memory.